test/e2e/utils_test.go

```diff
@@ -0,0 +1,77 @@
+//go:build e2e_tests
+// +build e2e_tests
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// namespace create a namespace for a test case, and clean it up after the test finishes.
+func namespace(t *testing.T) (*corev1.Namespace, func()) {
+	namespaceForTestCase, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name())
+	exitOnErr(err)
+
+	cleanup := func() {
+		assert.NoError(t, clusters.CleanupGeneratedResources(ctx, env.Cluster(), t.Name()))
+	}
+
+	return namespaceForTestCase, cleanup
+}
+
+const gatewayOperatorImageKustomizationContents = `
+images:
+- name: ghcr.io/kong/gateway-operator:main
+  newName: %v
+  newTag: '%v'
+`
+
+func setOperatorImage() error {
+	var image string
+	if imageLoad != "" {
+		image = imageLoad
+	} else {
+		image = imageOverride
+	}
+
+	if image == "" {
+		fmt.Println("INFO: use default image")
+		return nil
+	}
+
+	// TODO: deal with image names in format <host>:<port>/<repo>/<name>:[tag]
+	// e.g localhost:32000/kong/gateway-operator:xxx
+	parts := strings.Split(image, ":")
+	if len(parts) != 2 {
+		fmt.Printf("could not parse override image '%s', use default image\n", image)
+		return nil
+	}
+	imageName := parts[0]
+	imageTag := parts[1]
+
+	fmt.Println("INFO: use custom image", image)
+
+	// TODO: write back the kustomization file after test finished?
+	buf, err := os.ReadFile("../../config/default/kustomization.yaml")
+	if err != nil {
+		return err
+	}
+
+	// append image contents to replace image
+	fmt.Println("INFO: replacing image in kustomization file")
+	appendImageKustomizationContents := []byte(fmt.Sprintf(gatewayOperatorImageKustomizationContents, imageName, imageTag))
+	newBuf := append(buf, appendImageKustomizationContents...)
+	err = os.WriteFile("../../config/default/kustomization.yaml", newBuf, os.ModeAppend)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	return os.WriteFile("../../config/default/kustomization.yaml", newBuf, os.ModeAppend)
```

Apart from the suggestion I'm kind of worried about the implicit dependency on this file. Is this being used from elsewhere? 

Also, not trying to pick on the name here but `setOperatorImage` doesn't really say much about the side effect I've mentioned about. There's also no comment on it which might exacerbate the surprise effect.

Maybe we could write that new kustomization file to a temp file and pass its path around and then ensure at the end of the tests that we clean it up? This doesn't have to happen in this PR but I'd really like to see this change given what I've described above.

Reply:
Good point. I agree that it would be nice if we could make a tempfile for this purpose instead of overwriting files in the repository before we ship this, as inevitably someone is going to accidentally commit those changes at some point.

Reply:
fixed. For the second question, I am looking at kustomize to find out how to do it.